lib/grape/validations/params_scope.rb

```diff
@@ -254,10 +254,15 @@ def validate_value_coercion(coerce_type, values)
         return if values.is_a?(Proc)
         coerce_type = coerce_type.first if coerce_type.is_a?(Array)
         value_types = values.is_a?(Range) ? [values.begin, values.end] : values
+        value_types = convert_booleans(value_types) if coerce_type == Virtus::Attribute::Boolean
         if value_types.any? { |v| !v.is_a?(coerce_type) }
           fail Grape::Exceptions::IncompatibleOptionValues.new(:type, coerce_type, :values, values)
         end
       end
+
+      def convert_booleans(values)
```

Comment:
Since this is not reused and is a 1-liner I think we shouldn't make it a method. I know it feels long, but either roll it in, or make coerce_xyz for other types?